// uw-data/other/sysMsgCode.js
/**
 * Created by Administrator on 2015/4/9.
 */
var sysMsgCode = {
    //竞技场校验
    c_1_1:"c_1_1",
    c_1_2:"c_1_2",
    //副本
    c_2_1:"c_2_1",
    c_2_2:"c_2_2",
    c_2_3:"c_2_3",
    c_2_4:"c_2_4",
    c_2_5:"c_2_5",
    //装备
    c_3_1:"c_3_1",
    c_3_2:"c_3_2",
    //兑换
    c_4_1:"c_4_1",
    //英雄
    c_5_1:"c_5_1",
    c_5_2:"c_5_2",
    c_5_3:"c_5_3",
    c_5_4:"c_5_4",
    c_5_5:"c_5_5",
    c_5_6:"c_5_6",
    c_5_7:"c_5_7",
    c_5_8:"c_5_8",
    c_5_9:"c_5_9",
    c_5_10:"c_5_10",
    //物品
    c_6_1:"c_6_1",
    c_6_2:"c_6_2",
    //抽奖
    c_7_1:"c_7_1",

    //公会
    c_8_1: "c_8_1",
    c_8_2: "c_8_2",
    c_8_3: "c_8_3",
    c_8_4: "c_8_4",
    c_8_5: "c_8_5",
    c_8_6: "c_8_6",
    c_8_7: "c_8_7",
    c_8_8: "c_8_8",
    c_8_9: "c_8_9",
    c_8_10: "c_8_10",
    c_8_11: "c_8_11",
    c_8_12: "c_8_16",
    c_8_13: "c_8_13",
    c_8_14: "c_8_14",
    c_8_15: "c_8_15",
    c_8_16: "c_8_16",
    c_8_17: "c_8_17",
    c_8_18: "c_8_18",
    c_8_19: "c_8_19",
    c_8_20: "c_8_20",
    c_8_21: "c_8_21",
    c_8_22: "c_8_22",
    c_8_23: "c_8_23",
    c_8_24: "c_8_24",
    c_8_25: "c_8_25",
    c_8_26: "c_8_26",
    c_8_27: "c_8_27",
    c_8_28: "c_8_28",
    c_8_29: "c_8_29",
    c_8_30: "c_8_30",
    c_8_31: "c_8_31",
    c_8_32: "c_8_32",
    c_8_33:"c_8_33",
    c_8_34:"c_8_34",
    c_8_35:"c_8_35",
    c_8_36:"c_8_36",
    c_8_37:"c_8_37",
    c_8_38:"c_8_38",
    c_8_39:"c_8_39",
    c_8_40:"c_8_40",
    c_8_41:"c_8_41",
    c_8_42:"c_8_42",
    c_8_43:"c_8_43",
    c_8_44:"c_8_44",
    c_8_45:"c_8_45",
    c_8_46:"c_8_46",
    c_8_47:"c_8_47",
    c_8_48:"c_8_48",
    c_8_49:"c_8_49",
    c_8_50:"c_8_50",
    c_8_51:"c_8_51",
    c_8_52:"c_8_52",
    c_8_53:"c_8_53",
    c_8_54:"c_8_54",
    c_8_55:"c_8_55"
};

module.exports = sysMsgCode;
