// uw-view-http/controllers/pay/index.js
/**
 * Created by Administrator on 2015/4/16.
 */

var path = require("path");
var sdkBiz = require("uw-sdk").sdkBiz;
var logger = require('uw-log').getLogger("uw-pay", __filename);
exports.config ={
    method:"use"
};
exports.handler = function (req, res, next) {
    logger.debug("支付回调日志：req.body:%j",req.body);
    logger.debug("支付回调日志：req.query:%j",req.query);
    /*    var body = {
        "orderId":"42336448587846634645744363464A4248673D3D",
        "id":"3d22fddeb484b4458eac95eddb4de89d",
        "money":"6",
        "ext":"xyz",
        "time":"1430193421",
        "serverId":"1",
        "goodsId":"1",
        "goodsNumber":"1",
        "sign":"d721c8605ca8faaf3e632404ec27896c"
    };*/
    var orderData = {
        orderId: "",//订单号
        id: "",
        money: "",
        time: "",
        sign: "",
        ext: "",
        goodsId: "",
        goodsNumber: "",
        serverId: ""
    };
    if (req.body && req.body["orderId"]) {
        var body = req.body;
        orderData.orderId = body["orderId"];
        orderData.id = body["id"];
        orderData.money = body["money"];
        orderData.time = body["time"];
        orderData.sign = body["sign"];
        orderData.ext = body["ext"];
        orderData.goodsId = body["goodsId"];
        orderData.goodsNumber = body["goodsNumber"];
        orderData.serverId = body["serverId"];
    } else {
        orderData.orderId = req.query["orderId"];
        orderData.id = req.query["id"];
        orderData.money = req.query["money"];
        orderData.time = req.query["time"];
        orderData.sign = req.query["sign"];
        orderData.ext = req.query["ext"];
        orderData.goodsId = req.query["goodsId"];
        orderData.goodsNumber = req.query["goodsNumber"];
        orderData.serverId = req.query["serverId"];
    }
    sdkBiz.checkPay(orderData,function(err,data){
        if(err){
            logger.info("支付回调日志：失败！:%j req.body:%j",err,req.body);
            res.render("index", {data: "fail"});
        }else{
            logger.info("支付回调日志：成功！");
            res.render("index", {data: "success"});
        }
    });
};
